eyre/tests/test_location.rs

```diff
@@ -154,7 +158,7 @@ fn test_option_compat_wrap_err_with() {
     }));
 
     use eyre::ContextCompat;
-    let err = None::<()>.wrap_err_with(|| "oopsie").unwrap_err();
+    let err = None::<()>.with_context(|| "oopsie").unwrap_err();
```

Comment:
This correctly adheres to our model now as you can not "wrap an error" over an option.

You now either use the anyhow compatibility flag, or as the `AnyhowCompat` suggests the std methods like `ok_or_else` etc